Intersect.Server.Framework/Items/EntityItemSource.cs

```diff
@@ -0,0 +1,16 @@
+using Intersect.Enums;
+using Intersect.Server.Framework.Entities;
+
+namespace Intersect.Server.Framework;
+
+public class EntityItemSource: IItemSource
+{
+    public Guid Id { get; set; }
+    public EntityType EntityType { get; set; }
+    public WeakReference<IEntity> EntityReference { get; set; }
+}
+
+public class EntityItemSource<TEntity> : EntityItemSource where TEntity : class, IEntity
+{
+    public new WeakReference<TEntity> EntityReference { get; set; }
```

Comment:
Back this with a local variable, `set` should be `init` and `init` should set both the local variable and `base.EntityReference`.